### PR TITLE
Failing silently if syntax is unsupported

### PR DESCRIPTION
**Pull request by willful759**
This is my first commit to a open source project, I decided to tackle this specific issue: #1709 

Added the flag `fail-if-unsupported`, allows `bat` to fail without output and with a non-zero status code, so that it can be used as an input preprocessor

I'm new to rust and to contributing to open source so please give any feedback you have

### CODE REVIEW

**Review by keith-hall — COMMENTED** [review]
Thanks for your contribution! Looks good to me :)

The only feedback I have is maybe to add some [regression tests]( https://github.com/sharkdp/bat/blob/master/tests/integration_tests.rs) if possible. (And to think about how we want it to work if there are multiple input files and some have a syntax mapping and some don't.)

It may also be nice to make the changelog entry a bit more descriptive about what "unsupported" means - basically just like the help entry.
Something we always forget is to update the [`man` page](https://github.com/sharkdp/bat/blob/master/assets/manual/bat.1.in) when adding/updating command line args - it would be great if you could add it there too please.

**Comment by keith-hall** [comment]
Also, I would say, don't be afraid to force-push to tidy up the commit history (and fix typos) - at least before anyone has reviewed your changes :) Don't worry about it now, though - I think it may make sense to squash merge, but we'll see what the other maintainers say :)

**Comment by willful759** [comment]
Thanks for the feedback! iḿ also quite inexperienced with git, I've only used personally, but I will look forward into your suggestions

In the case we have multiple input files and only some of them don't have specific syntax we can use, I would personally just let it fail with a different error code, but I would really appreciate community suggestions

File: tests/integration_tests.rs [hunk 1]
```diff
@@ -1312,3 +1312,14 @@ fn acknowledgements() {
         )
         .stderr("");
 }
+
+// Regression test for https://github.com/sharkdp/bat/issues/1709
+#[test]
+fn fail_if_unsupported_supported() {
```
Comment (Enselic) [inline tests/integration_tests.rs:1318]: Thank you for taking the time to make a contribution!

Before a detailed review, it will be very useful to have a broad set of regression tests in place. The test you added is only for the positive case. We would need tests also for the negative case, i.e. when the exit code is not 0.

I am also a bit skeptical about the name `--fail-if-unsupported` because it does not make much sense in and of itself. Since the use case specifically is usage as a less input preprocessor, I think we can simply call the flag `--as-input-processor-for-less`. I think it is perfectly fine to have long names for "experimental" feature. If it turns out that the flag can by used more broadly, we can also broaden the flag to e.g. just `--as-input-preprocessor`.

It is better to start narrow and iteratively broaden, than to start broad and iteratively narrow. Because the latter is hard to do in a backwards compatible way.

**Comment by Enselic** [comment]
Closing due to inactivity to keep he PR inbox clean, but of course feel free to re-open if you resume work on this!

**Comment by mnp** [comment]
Came here looking for this feature and was sad it didn't get completed.
